base/src/consensus_state.rs

```diff
@@ -80,6 +93,36 @@ pub struct ConsensusState {
     pub supercharge_coinbase: bool,
 }
 
+impl Hashable for ConsensusState {
```

Comment:
This impl block is mostly same except for the function body of `to_roinput`. Can we write a readable macro to remove the repetitive parts? I believe you could just use a token tree `tt` for the function body.